src/_pytest/runner.py

```diff
@@ -212,6 +225,8 @@ class CallInfo:
     start = attr.ib()
     stop = attr.ib()
     when = attr.ib()
+    item = attr.ib(type=Optional[Node])
+    _report = attr.ib(type=Optional[TestReport], default=None)
```

Comment:
from my pov this brings together concepts that shouldn't necessary be merged directly,
past experiences of evolving pytest using similar re-factorings tended to demonstrate huge costs in the aftermath (see the old mark system, next-item and fixture setupstate, collection tree structure)

i am afraid that this change here could manifest a similar type of cost later on (but i also hope that i'm wrong with that hunch)